### PR TITLE
Open-source documentation

### PR DESCRIPTION
**Pull request by dominiklohmann**
This PR is revamping our docs. 

### :memo: Checklist

- [x] Integrate docs build into CI (@dominiklohmann)
- [x] Revamp landing page (@mavam)
- [x] Add scaffold with content (@mavam)
- [x] Tune CSS (@mavam)
- [x] Decide what to do with CLI docs (@mavam)
- [x] Fix favicon (@mavam)
- [x] Fix Style Check workflow (@dominiklohmann)
- [x] ~Squash noise and ship~
- [x] Fix dependabot (@dominiklohmann)

### :dart: Review Instructions

Commit-by-commit.

### CODE REVIEW

**Comment by mavam** [comment]
[sc-32870]

**Comment by mavam** [comment]
@dominiklohmann any ideas why this triggers clang-format?

**Comment by mavam** [comment]
@dominiklohmann handing the edit token to you to fix the clang-format style check.

**Review by mavam — APPROVED** [review]
(I have to approve this myself since @dominiklohmann created the PR and I took it over.)

**Comment by dominiklohmann** [comment]
> EDIT: if you think it was a mistake to remove the CLI docs, feel free to drop the last commit.

It was, that breaks the `vast <command> documentation` and `vast manual` commands. I dropped the commit.

> EDIT2: you may want to double check the publishing section for deploying GitHub Pages here: https://v1.docusaurus.io/docs/ro/publishing#using-github-pages.

Ours is based off of the same docs page from the docusaurus v2 docs.

> Other than that, we're ready to roll with this scaffold and can merge it.

Let's wait until after Easter, I'd like to go over all your changes first.

> @dominiklohmann handing the edit token to you to fix the clang-format style check.

Done and pushed.

Regarding the squashing, I don't even think we need to. It's not that much, and I like preserving the history of how we got here.

File: .github/dependabot.yml [hunk 1]
```diff
@@ -9,3 +9,7 @@ updates:
     directory: "/"
     schedule:
       interval: "daily"
+  - package-ecosystem: "yarn"
```
Comment (mavam) [inline .github/dependabot.yml:12]:
```suggestion
  - package-ecosystem: "npm"
```

The error we got was:
> The property '#/updates/1/package-ecosystem' value "yarn" did not match one of the following values: npm, bundler, composer, maven, mix, cargo, gradle, nuget, gomod, docker, elm, gitsubmodule, github-actions, pip, terraform, pub

**Comment by mavam** [comment]
It looks like the builds for the static binary are now failing, too. As if the C++ headers that contain the Markdown for the CLI commands are no longer present.

**Comment by dominiklohmann** [comment]
Fixed both the static build and the Dependabot setup. I think we can merge this regardless of the feature freeze as it's entirely independent of VAST.

**Comment by dominiklohmann** [comment]
@mavam one thing yet to fix: The page title of all directory README pages is `README | VAST`, rather than, say, `Get Started | VAST` for `/vast/docs/start/`.

Also, should we align the URLs with the actual titles, e.g., consistently use `/vast/docs/get-started/` if the page title is `Get Started`?

**Pull request merged**